py_mmd_tools/mmd_to_nc.py

```diff
@@ -0,0 +1,256 @@
+#!/usr/bin/env python3
+"""
+Tool for parsing an XML metadata file, following the MET Norway Metadata specification (MMD),
+ and updating an NetCDF file, following the Attribute Convention for Data Discovery (ACDD).
+
+License:
+
+This file is part of the py-mmd-tools repository
+<https://github.com/metno/py-mmd-tools>.
+
+py-mmd-tools is licensed under the Apache License 2.0
+<https://github.com/metno/py-mmd-tools/blob/master/LICENSE>
+"""
+
+import yaml
+import netCDF4 as nc
+import lxml.etree as ET
+import py_mmd_tools
+from pkg_resources import resource_string
+
+
+class Mmd_to_nc(object):
+
+    def __init__(self, mmd_product, nc_file):
+        """Class for updating a NetCDF file that is compliant with the CF-conventions and ACDD
+        from an MMD XML file.
+        Args:
+            mmd_product (str): Input MMD xml file.
+            nc_file (pathlib): Nc file to update.
+        """
+        super(Mmd_to_nc, self).__init__()
+
+        # NC file
+        self.nc = nc_file
+
+        # MMD file content / namespaces
+        tree = ET.parse(mmd_product)
+        self.tree = tree
+        self.namespaces = tree.getroot().nsmap
+
+        # Get translation file between MMD and ACDD
+        self.mmd_yaml = yaml.load(
+            resource_string(py_mmd_tools.__name__, 'mmd_elements.yaml'), Loader=yaml.FullLoader
+        )
+        return
+
+    @staticmethod
+    def get_acdd(t):
+        """
+        Get acdd or acdd_ext value from a dictionary.
+        """
+        if 'acdd' in t:
+            out = t['acdd']
+        elif 'acdd_ext' in t:
+            out = t['acdd_ext']
+        else:
+            out = None
+        try:
+            sep = t['separator']
+        except KeyError:
+            sep = None
+        return out, sep
+
+    @staticmethod
+    def process_elem(elem, elem_info, tag):
+        out = {}
+        sep = None
+        if not elem_info[tag] is None:
+            acdd_name, sep = Mmd_to_nc.get_acdd(elem_info[tag])
+            if acdd_name is not None:
+                out[acdd_name] = elem.text
+        return out, sep
+
+    @staticmethod
+    def update_acdd(dict1, dict2, sep=None):
+        # First time = 'initialize' the dictionary
+        if len(dict1) == 0:
+            return dict2
+        # If dict2 is empty, no update
+        elif dict2 is None:
+            return dict1
+        # Main case
+        else:
+            # Check if key already present in dict1
+            for key in dict2:
+                if key in dict1:
+                    # If so, it must be a list, so we append it
+                    try:
+                        dict1[key] = sep[key].join([dict1[key], dict2[key]])
+                    except TypeError:
+                        dict1[key] = sep.join([dict1[key], dict2[key]])
+                else:
+                    dict1[key] = dict2[key]
+        return dict1
+
+    def get_last_metadata_update(self, element):
```

Comment:
Not sure if I understand this function. Can you explain, and perhaps add a unit test to show the use case?